source/agora/utils/Test.d

```diff
@@ -589,3 +591,174 @@ unittest
+    foreach (idx, ref in_; result.inputs)
+    {
+        auto rng = from.find!(a => a.address == input.outputs[idx].address);
+        assert(rng.length);
+        const owner = rng[0];
+        in_.signature = sign(owner, resultHash);
+    }
+    return result;
+}
+
+/// Test for a split with the same amount of outputs as inputs
+/// Essentially doing an equality transformation
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(8).map!(_ => KeyPair.random()).array;
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 8 txs, hence it's just equality
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 8);
+    // Since the amount is evenly distributed in Genesis,
+    // they all have the same value
+    const ExpectedAmount = first.outputs[0].value;
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with twice as many outputs as inputs
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(16).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 16 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 16);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(16);
+    assert(remainder == Amount(0));
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with remainder
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(3).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 3 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 3);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(3);
+    assert(remainder != Amount(0));
+
+    // The first output includes the remainder.
+    assert(equalTx.outputs[0].value >= ExpectedAmount);
+    assert(equalTx.outputs[1].value == ExpectedAmount);
+    assert(equalTx.outputs[2].value == ExpectedAmount);
+}
+
+/// Test with one output key
+unittest
+{
+    import std.range;
+
+    KeyPair key = KeyPair.random();
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], [key.address]);
+    // This transaction has 3 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 1);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
```

Comment:
This usage of `getSumOutput` is okay-ish, but the return value needs to be checked!

```suggestion
    Amount ExpectedAmount;
    assert(first.getSumOutput(ExpectedAmount));
```